src/main/kotlin/org/wycliffeassociates/otter/common/domain/resourcecontainer/ImportResourceContainer.kt

```diff
@@ -4,85 +4,139 @@ import io.reactivex.Single
 import io.reactivex.schedulers.Schedulers
 import org.wycliffeassociates.otter.common.collections.tree.Tree
 import org.wycliffeassociates.otter.common.data.model.Collection
+import org.wycliffeassociates.otter.common.domain.resourcecontainer.project.IZipEntryTreeBuilder
 import org.wycliffeassociates.otter.common.domain.resourcecontainer.project.IProjectReader
 import org.wycliffeassociates.otter.common.persistence.IDirectoryProvider
 import org.wycliffeassociates.otter.common.persistence.repositories.IResourceContainerRepository
+import org.wycliffeassociates.resourcecontainer.DirResourceContainer
 import org.wycliffeassociates.resourcecontainer.ResourceContainer
 import java.io.File
 import java.io.IOException
+import java.util.zip.ZipFile
 
 class ImportResourceContainer(
-        private val resourceContainerRepository : IResourceContainerRepository,
-        private val directoryProvider: IDirectoryProvider
+        private val resourceContainerRepository: IResourceContainerRepository,
+        private val directoryProvider: IDirectoryProvider,
+        private val zipEntryTreeBuilder: IZipEntryTreeBuilder
 ) {
     fun import(file: File): Single<ImportResult> {
         return when {
             file.isDirectory -> importContainerDirectory(file)
+            file.extension == "zip" -> importContainerZipFile(file)
             else -> Single.just(ImportResult.INVALID_RC)
         }
     }
 
+    private fun File.contains(name: String): Boolean {
+        if (!this.isDirectory) {
+            throw Exception("Cannot call contains on non-directory file")
+        }
+        return this.listFiles().map { it.name }.contains(name)
+    }
+
+    private fun importContainerZipFile(file: File): Single<ImportResult> {
+        val zip = ZipFile(file)
+        zip.getEntry("manifest.yaml")
+                ?: return Single.just(ImportResult.LOAD_RC_ERROR)
```

Comment:
It'd be very convenient if this could go one folder deeper to find the manifest.  Then, a user could grab an RC from DCS or WACS by clicking "Download this repository" (which creates an archive with the RC root as a folder inside). https://content.bibletranslationtools.org/WycliffeAssociates-mirror/en_ulb

Reply:
I believe that's based on how the user extracts the repo; it extracts the appropriate folder, but if they extract to a folder, then you run into this issue

Reply:
I'm suggesting that doing no extraction at all would be easiest. Just click that download button on the WACS project main screen and get the zip. That zip has everything pushed down one level, inside a en_tn folder.

Since that's the most obvious source of zipped RCs, it'd be handy if a user (or maybe the auto-downloader in the future) could use those without any other manipulation.

Reply:
Spawned issue https://github.com/WycliffeAssociates/otter-jvm/issues/425